lib/collection/src/operations/types.rs

```diff
@@ -995,66 +995,33 @@ pub struct VectorParamsDiff {
     pub hnsw_config: Option<HnswConfigDiff>,
 }
 
-/// Vector update params separator for single and multiple vector modes
-///
-/// Single mode:
-///
-/// { "hnsw_config": { "m": 8 } }
-///
-/// or multiple mode:
+/// Vector update params for multiple vectors
 ///
 /// {
-///     "default": {
+///     "vector_name": {
 ///         "hnsw_config": { "m": 8 }
 ///     }
 /// }
 #[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq, Hash, Eq)]
-#[serde(rename_all = "snake_case", untagged)]
-pub enum VectorsConfigDiff {
-    Single(VectorParamsDiff),
-    Multi(BTreeMap<String, VectorParamsDiff>),
-}
-
-impl VectorsConfigDiff {
-    pub fn get_params(&self, name: &str) -> Option<&VectorParamsDiff> {
-        match self {
-            VectorsConfigDiff::Single(params) => (name == DEFAULT_VECTOR_NAME).then_some(params),
-            VectorsConfigDiff::Multi(params) => params.get(name),
-        }
-    }
-
-    /// Iterate over the named vector parameters.
-    ///
-    /// If this is `Single` it iterates over a single parameter named [`DEFAULT_VECTOR_NAME`].
-    pub fn params_iter<'a>(&'a self) -> Box<dyn Iterator<Item = (&str, &VectorParamsDiff)> + 'a> {
-        match self {
-            VectorsConfigDiff::Single(p) => Box::new(std::iter::once((DEFAULT_VECTOR_NAME, p))),
-            VectorsConfigDiff::Multi(p) => Box::new(p.iter().map(|(n, p)| (n.as_str(), p))),
-        }
-    }
-}
+pub struct VectorsConfigDiff(pub BTreeMap<String, VectorParamsDiff>);
```

Comment:
Empty string as default vector name is an internal implementation details. To resolve single-multivector configs, I propose to add field `vector: Option<VectorParamsDiff>` into `UpdateCollection` structure. For multiple vectors user set `vectors` field and for single `vector` field

Reply:
> Empty string as default vector name is an internal implementation details.

You're right.

While using `vector` and `vectors` is possible, I think it may end up being more confusing than useful. We've seen reports on typo's like that before.

Reply:
If we have typo's about it, we can rename `vector` -> `single_vector` and `vectors` -> `multiple_vectors`. But I don't like this option

Reply:
I guess most users use single-vector config and may not know about multiple vector support. and I expect that `""` is confusing

Reply:
> I guess most users use single-vector config and may not know about multiple vector support. and I expect that `""` is confusing

Fair point.

Though if users use a single vector, they better update the collection HNSW/quantization parameters, rather than the vector one. In that case no `""` name is needed at all.

In other words, rather than doing this:

```json
{
    "vectors": {
        "": {
            "hnsw_config": {
                "m": 32
            }
        }
    }
}
```

They better just do this:

```json
{
    "hnsw_config": {
        "m": 32
    }
}
```

Reply:
:+1: 

Yeah it is still not perfect. But I'm afraid we have to cut _some_ corners to resolve this. This is the best I could come up with.